document/JS.bs

```diff
@@ -0,0 +1,935 @@
+  The WebAssembly.instantiate(|moduleObject|, |importObject|) asynchronously instantiates a compiled module.
+-->
+
+<div algorithm>
+  A WebAssembly module AST |module| is said to be <dfn>valid</dfn> if:
+    1. [=validate_module=](_module_) is not [=error=], and
+    1. The UTF-8 encoding of all <code>name</code> productions in the binary format, including in a custom <code>name</code> section if present, is valid.
+</div>
+
+<div algorithm>
+  The <dfn method for="WebAssembly">validate(bytes)</dfn> method, when invoked, performs the following steps:
+    1. Let |module| be [=decode_module=](_bytes_). If |module| is [=error=], return false.
+    1. If |module| is not [=valid=], return false.
+    1. Return true.
+</div>
+
+<div algorithm>
+  To <dfn>compile a WebAssembly module</dfn> from a {{BufferSource}} |bytes|, perform the following steps:
+    1. Let |module| be [=decode_module=](_bytes_). If |module| is [=error=], return the string "CompileError".
+    1. If |module| is not [=valid=], return the string "CompileError"
+    1. Return |module|
+</div>
+
+<div algorithm>
+  To <dfn>construct a WebAssembly module object</dfn> from a module |module|, perform the following steps:
+
+    1. Let |moduleObject| be a new {{Module}} object.
+    1. Set |moduleObject|.\[[Module]] to |module|.
+    1. Set |moduleObject|.\[[Bytes]] to |bytes|.
+    1. Return |moduleObject|.
+</div>
+
+<div algorithm>
+  To <dfn>asynchronously compile a WebAssembly module</dfn> from a {{BufferSource}} |bytes|, with the promise |promise|, using optional [=task source=] |taskSource|, perform the following steps:
+
+    1. In parallel, [=compile a WebAssembly module|compile the WebAssembly module=] |bytes| and store the result as |module|.
+    1. When the above operation completes, queue a task to perform the following steps. If |taskSource| was provided, queue the task on that task source.
+        1. If |module| is the string "CompileError", throw a {{CompileError}} exception.
+        1. Otherwise,
+            1. [=Construct a WebAssembly module object=] from |module| and let |moduleObject| be the result.
+            1. [=Resolve=] |promise| with |moduleObject|.
+</div>
+
+<div algorithm>
+    The <dfn method for="WebAssembly">compile(bytes)</dfn> method, when invoked, performs the following steps:
+    1. Let |stableBytes| be a [=get a copy of the buffer source|copy of the bytes held by the buffer=] |bytes|.
+    1. Let |promise| be [=a new promise=].
+    1. [=Asynchronously compile a WebAssembly module=] from |stableBytes| with |promise|.
+    1. Return |promise|.
+</div>
+
+<div algorithm>
+  To <dfn>instantiate a WebAssembly module</dfn> from a {{Module}} |moduleObject| and imports |importObject|, perform the following steps:
+    1. Let |module| be |moduleObject|.\[[Module]].
+    1. If |module|.imports is not an empty list, and |importObject| is undefined, throw a [=TypeError=] exception.
```

Comment:
`{{TypeError}}`, not `[=TypeError=]`

Reply:
Really? The JS one gives much more information, and they are referring to the same thing.

Reply:
Oh, I didn't realize that {{TypeError}} went somewhere else. In that case, add something to the links block to make it go to the JS spec, but then use {{TypeError}} so you get the more consistent typography.

Reply:
That is prettier! Thanks.